Updates for CITP Pioneer study schema

akohlbre opened the pull request:
Main changes are two new fields added to the schema (also removed overly-aggressive validation).

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] If the PR comes from a fork, trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional) and review the report posted in the comments.

For glean changes:
- [ ] Update `templates/include/glean/CHANGELOG.md`

acmiyaguchi reviewed (changes requested):
Hi @akohlbre, could you rebase this PR onto the latest version of mozilla-pipeline-schemas. Make sure the main repo is set as a remote (e.g. `upstream`) and rebase the new patches onto the main branch. It should be something like the following:

```bash
git rebase --onto upstream/master HEAD~2
```

acmiyaguchi commented on schemas/pioneer-citp-news-disinfo/measurements/measurements.1.schema.json:
```diff
@@ -189,6 +168,9 @@
       "type": "object"
+    },
+    "WebScience.SurveyId": {
+      "type": "string"
```
It's good practice to document the use of this field using the `description` property, which will get inserted into the BigQuery schema.

acmiyaguchi commented on schemas/pioneer-citp-news-disinfo/measurements/measurements.1.schema.json:
```diff
@@ -148,21 +136,15 @@
-                        "other"
-                      ],
-                      "type": "string"
+                      "type": "integer"
```
This type change won't play well with automated checks in our deployment pipeline. There are two ways to change this. The first is document this field as deprecated in a description and to add a new property (e.g. `pageClassification`) which would be evolution compatible. The other is to copy the schema with the version bumped.

akohlbre commented:
Alright --
>could you rebase this PR onto the latest version of mozilla-pipeline-schemas

Oops, sorry, I totally forgot to grab updates. Think I got the rebase to work, thanks.

>It's good practice to document the use of this field using the description property, which will get inserted into the BigQuery schema.

Added a description for the survey ID field.

>This type change won't play well with automated checks in our deployment pipeline. There are two ways to change this. The first is document this field as deprecated in a description and to add a new property (e.g. pageClassification) which would be evolution compatible. The other is to copy the schema with the version bumped.

Added the old field back and labeled it deprecated, and used your suggested name for the replacement.

Thanks!

acmiyaguchi commented:
Looks good to me.
